example/apache-ignite/src/main/scala/org/finos/vuu/example/ignite/IgniteOrderStore.scala

```diff
@@ -58,25 +59,17 @@ class IgniteOrderStore(private val parentOrderCache: IgniteCache[Int, ParentOrde
     parentOrderCache.get(id)
   }
 
-  def findChildOrder(sqlFilterQueries: String, sqlSortQueries: String, rowCount: Int, startIndex: Long): Iterable[ChildOrder] = {
+  def findChildOrder(sqlFilterQueries: String, sqlSortQueries: String, rowCount: Int, startIndex: Long): LazyList[ChildOrder] = {
     val whereClause = if(sqlFilterQueries == null || sqlFilterQueries.isEmpty) "" else s" where $sqlFilterQueries"
     val orderByClause = if(sqlSortQueries == null || sqlSortQueries.isEmpty) " order by id" else s" order by $sqlSortQueries"
     val query = new SqlFieldsQuery(s"select * from ChildOrder$whereClause$orderByClause limit ? offset ?")
     query.setArgs(rowCount, startIndex)
 
-    val results = childOrderCache.query(query)
-
-    var counter = 0
-    val buffer = mutable.ListBuffer[ChildOrder]()
-    results.forEach(item => {
-      buffer.addOne(toChildOrder(item))
-      counter += 1
-    })
-
-    logger.info(s"Loaded Ignite ChildOrder for $counter rows, from index : $startIndex where $whereClause order by $sqlSortQueries")
+    val results = LazyList.from(childOrderCache.query(query).asScala).map(i => toChildOrder(i.asScala.toList))
```

Comment:
Not sure if it achieves the same, but would be ideal if we can take advantage of that as well. I'll look into this.

My thinking was mostly around to delay the `List` ->`toChildOrder` -> `toRowData` conversion until actually needed, which in this case would be when we update the table row.

Before we were iterating through the query result and adding it to a ListBuffer, so that wasn't taking advantage of the lazy load from the query cursor anyways.